constraints/miden-vm/decoder.air

```diff
@@ -0,0 +1,493 @@
+fn f_loop(b: vector[7]) -> scalar:
+    return b[6] * binary_not(b[5]) * b[4] * b[3] * b[2] * b[1]
+
+
+# Returns the f_repeat operation flag which is set when REPEAT operation is executed.
+fn f_repeat(b: vector[7], extra: scalar) -> scalar:
+    return extra * b[4] * binary_not(b[3]) * b[2]
+
+
+# Returns the f_repeat operation flag which is set when REPEAT operation is executed in the next 
+# row.
+fn f_repeat_next(b: vector[7], extra: scalar) -> scalar:
+    return extra' * b[4]' * binary_not(b[3]') * b[2]'
+    
+
+# Returns the f_span operation flag which is set when SPAN operation is executed.
+fn f_span(b: vector[7]) -> scalar:
+    return b[6] * binary_not(b[5]) * b[4] * b[3] * binary_not(b[2]) * binary_not(b[1])
+
+
+# Returns the f_respan operation flag which is set when RESPAN operation is executed.
+fn f_respan(b: vector[7], extra: scalar) -> scalar:
+    return extra * b[4] * b[3] * binary_not(b[2])
+
+
+# Returns the f_respan operation flag which is set when RESPAN operation is executed in the next 
+# row.
+fn f_respan_next(b: vector[7], extra: scalar) -> scalar:
+    return extra' * b[4]' * b[3]' * binary_not(b[2]')
+
+
+# Returns the f_call operation flag which is set when CALL control operation is executed.
+fn f_call(b: vector[7], extra: scalar) -> scalar:
+    return extra * binary_not(b[4]) * b[3] * b[2]
+
+
+# Returns the f_syscall operation flag which is set when SYSCALL control operation is executed.
+fn f_syscall(b: vector[7], extra: scalar) -> scalar:
+    return extra * binary_not(b[4]) * b[3] * binary_not(b[2])
+
+
+# Returns the f_end operation flag which is set when END operation is executed.
+fn f_end(b: vector[7], extra: scalar) -> scalar:
+    return extra * b[4] * binary_not(b[3]) * binary_not(b[2])
+
+
+# Returns the f_end operation flag which is set when END operation is executed in the next 
+# row.
+fn f_end_next(b: vector[7], extra: scalar) -> scalar:
+    return extra' * b[4]' * binary_not(b[3]') * binary_not(b[2]')
+
+
+# Returns the f_halt operation flag which is set when HALT operation is executed.
+fn f_halt(b: vector[7], extra: scalar) -> scalar:
+    return extra * b[4] * b[3] * b[2]
+
+
+# Returns the f_halt operation flag which is set when HALT operation is executed in the next 
+# row.
+fn f_halt_next(b: vector[7], extra: scalar) -> scalar:
+    return extra' * b[4]' * b[3]' * b[2]'
+
+
+# Returns the f_push operation flag which is set when PUSH operation is executed.
+fn f_push(b: vector[7], extra: scalar) -> scalar:
+    return extra * binary_not(b[4]) * binary_not(b[3]) * b[2]
+
+
+# Returns the f_ctrl flag which is set when any one of the control flow operations that has flags 
+# above (JOIN, SPLIT, LOOP, CALL, SYSCALL) is being executed.
+fn f_ctrl(b: vector[7], extra: scalar) -> scalar:
+    # flag for span, join, split, loop
+    let f_sjsl = b[6] * binary_not(b[5]) * b[4] * b[3]
+    # flag for end, repeat, respan, halt
+    let f_errh = b[6] * b[5] * b[4]
```

Comment:
nit: Would be great to have a blank line here
question: Should we capitalize SPAN, JOIN, etc in the comments?